src/boosting/gbdt.cpp

```diff
@@ -269,6 +275,16 @@ void GBDT::RefitTree(const std::vector<std::vector<int>>& tree_leaf_prediction)
   CHECK_EQ(static_cast<size_t>(models_.size()), tree_leaf_prediction[0].size());
   int num_iterations = static_cast<int>(models_.size() / num_tree_per_iteration_);
   std::vector<int> leaf_pred(num_data_);
+  if (linear_tree_) {
+    int max_leaves = 0;
+    for (size_t i = 0; i < tree_leaf_prediction.size(); ++i) {
+      for (size_t j = 0; j < tree_leaf_prediction[i].size(); ++j) {
+        max_leaves = std::max(max_leaves, tree_leaf_prediction[i][j]);
+      }
+    }
+    max_leaves += 1;
```

Comment:
why `+1` here?

Reply:
The maximum of tree_leaf_prediction is 1 less than the number of leaves in the tree (since leaf predictions are labelled 0, 1, ..., num_leaves - 1).